src/serializer/ResidualReactElementSerializer.js

```diff
@@ -96,7 +96,7 @@ export class ResidualReactElementSerializer {
           t.callExpression(funcId, originalCreateElementIdentifier ? [originalCreateElementIdentifier] : [])
         )
       );
-      this.residualHeapSerializer.emitter.emit(statement);
+      this.residualHeapSerializer.emitter.emitLazyReactElementInitializer(statement);
```

Comment:
If this intends to do what I think it does, then you want to do the following:
1. Pass the result of `this.residualHeapSerializer.isReferencedOnlyByAdditionalFunction(value)` (currently evaluated by caller) into this function; let's call it `additionalFunction`. It should be defined by construction.
2. Then replace this line with the following:
```js
this.residualHeapSerializer._getPrelude(additionalFunction).push(statement);
```

Reply:
That's exactly what I want to do!